config/cht/targets.js

```diff
@@ -0,0 +1,187 @@
+const extras = require('./nools-extras');
+const FIVE_YEARS_IN_DAYS = 1826;
+const {
+  today,
+  addDays,
+  isAlive,
+  getSubsequentPregnancyFollowUps,
+  isOnSameMonth,
+  getMostRecentLMPDateForPregnancy,
+  isActivePregnancy,
+  countANCFacilityVisits
+} = extras;
+
+module.exports = [
+
+  {
+    id: 'newborn-mortality',
+    type: 'percent',
+    icon: 'icon-death-neonatal',
+    goal: 0,
+    translation_key: 'targets.anc.newborn_mortality.title',
+    subtitle_translation_key: 'targets.all_time.subtitle',
+    appliesTo: 'contacts',
+    appliesToType: ['person'],
+    appliesIf: function (c) {
+      //console.log('a', c.contact.name, c.contact.date_of_birth, addDays(c.contact.date_of_birth, FIVE_YEARS_IN_DAYS).getTime(), today, addDays(c.contact.date_of_birth, FIVE_YEARS_IN_DAYS) > today);
+      return c && c.contact.date_of_birth && addDays(c.contact.date_of_birth, FIVE_YEARS_IN_DAYS) > today;
+    },
+    passesIf: function (c) {
+      //console.log('p', c.contact.name, c.contact.date_of_death, addDays(c.contact.date_of_birth, FIVE_YEARS_IN_DAYS) > new Date(c.contact.date_of_death));
+      return c.contact.date_of_death && addDays(c.contact.date_of_birth, FIVE_YEARS_IN_DAYS) > new Date(c.contact.date_of_death);
+    },
+    date: 'now',
+    idType: 'contact'
+  },
+
+  {
+    id: 'deaths-this-month',
+    type: 'count',
+    icon: 'icon-death-general',
+    goal: 0,
+    translation_key: 'targets.death_reporting.deaths.title',
+    subtitle_translation_key: 'targets.this_month.subtitle',
+    appliesTo: 'contacts',
+    appliesToType: ['person'],
+    appliesIf: function (c) {
+      return !isAlive(c) && isOnSameMonth(today, c.contact.date_of_death);
+    },
+    date: 'now'//'reported' does not work with contact here, because contact.reported_date can be older
+  },
+
+  // ANC: New Pregnancies as a count - this month with LMP, with a goal of 20
+  {
+    id: 'pregnancy-registrations-this-month',
+    type: 'count',
+    icon: 'icon-pregnancy',
+    goal: 20,
+    translation_key: 'targets.anc.new_pregnancy_registrations.title',
+    subtitle_translation_key: 'targets.this_month.subtitle',
+    appliesTo: 'reports',
+    appliesToType: ['pregnancy'],
+    appliesIf: function (c, r) {
+      if (r === null) return false;
```

Comment:
You do a lot of these specific "null checks" through your code. Generally speaking, the more generic check for "falsy" is typically more robust than specifically checking for the null condition. Why not `if (!r) return false;`? It seems like an improvement to replace all null checks throughout the code with the same sort of falsy check.

Reply:
Thanks, I have replaced null checks with falsy checks everywhere now.